rdflib/namespace/__init__.py

```diff
@@ -346,37 +346,27 @@ def __init__(self, graph: "Graph"):
         for p, n in self.namespaces():  # self.bind is not always called
             insert_trie(self.__trie, str(n))
 
-        # DefinedNamespace bindings.
-        self.bind("brick", BRICK)
-        self.bind("csvw", CSVW)
-        self.bind("dc", DC)
-        self.bind("dcat", DCAT)
-        self.bind("dcmitype", DCMITYPE)
-        self.bind("dcterms", DCTERMS)
-        self.bind("dcam", DCAM)
-        self.bind("doap", DOAP)
-        self.bind("foaf", FOAF)
-        self.bind("odrl", ODRL2)
-        self.bind("geo", GEO)
-        self.bind("org", ORG)
-        self.bind("owl", OWL)
-        self.bind("prof", PROF)
-        self.bind("prov", PROV)
-        self.bind("qb", QB)
-        self.bind("rdf", RDF)
-        self.bind("rdfs", RDFS)
-        self.bind("schema", SDO)
-        self.bind("sh", SH)
-        self.bind("skos", SKOS)
-        self.bind("sosa", SOSA)
-        self.bind("ssn", SSN)
-        self.bind("time", TIME)
-        self.bind("vann", VANN)
-        self.bind("void", VOID)
-        self.bind("xsd", XSD)
-
-        # Namespace bindings.
-        self.bind("xml", XMLNS)
+        # bind Namespaces as per options.
+        # default is core
+        if bind_namespaces == "core":
+            # bind a few core RDF namespaces
+            for prefix, ns in NAMESPACE_PREFIXES_CORE.items():
+                self.bind(prefix, ns)
+        elif bind_namespaces == "rdflib":
+            # bind all the Namespaces shipped with RDFLib
+            for prefix, ns in NAMESPACE_PREFIXES_RDFLIB.items():
+                self.bind(prefix, ns)
+            # ... don't forget the core ones too
+            for prefix, ns in NAMESPACE_PREFIXES_CORE.items():
+                self.bind(prefix, ns)
+        elif bind_namespaces == "cc":
+            # bind any prefix that can be found with lookups to prefix.cc
+            # first bind core and rdflib ones
+            # work out remainder - namespaces without prefixes
+            # only look those ones up
+            raise NotImplementedError("Haven't got to this option yet")
+        else:  # bind_namespaces is None
+            pass  # bind nothing
```

Comment:
May be a bit better to rather distinguish between explicit `None` and invalid/unknown string so that sending the wrong string results in a failure, will make a PR against your branch to show you what I mean.

Reply:
I think I get it. I haven't finished all the implementation here of course, so was going to add checks for `None` and other strings too but just haven't got to it so far.

Feel free to put in a PR or commits to the branch!

Reply:
PR here: https://github.com/RDFLib/rdflib/pull/1693